packages/core/src/request.ts

```diff
@@ -250,15 +256,32 @@ export class Request<UserData extends Dictionary = Dictionary> {
         (this.userData as Dictionary).label = value;
     }
 
+    /** Maximum number of retries for this request. Allows to override the global `maxRequestRetries` option of `BasicCrawler`. */
+    get maxRetries(): number {
+        return this.userData.__crawlee?.maxRetries;
+    }
```

Comment:
A bit off-topic, but how is this not a TS error - the optional access with `?.` can cause returning a nullish value (and the function signature allows only `number`), no?

Reply:
Interesting, the return type should include `undefined`.

Reply:
Btw it works because the `userData` is a dict of `any`, so is the `__crawlee` object.

Reply:
Ah right, thank you - although I would expect the compiler to say at least something - looks like a problematic cast to me to say the least 🤔

Reply:
The thing is that optional chaining on something that is not optional is not adding the `undefined` to the union type (which sounds correct to me), and as `any` will eat all the unions (`any | undefined` is again just `any`), this is what you get.

Reply:
That part I get, it's more about the `any -> number` for me, a bit like the meme with fitting all the shapes in the square hole. But I'll have to learn to live with that, I guess 🤷🏼😄